rbac/common/base/base_address.py

```diff
@@ -116,5 +217,38 @@ def address_is(self, address):
 
     def addresses_are(self, addresses):
         """Determines if all addresses given are of the classes' address type"""
-        # pylint: disable=unnecessary-lambda
-        return all(map(lambda a: self.address_is(a), addresses))
+        return all([self.address_is(a) for a in addresses])
+
+    def get_address_type(self, address):
+        """Returns the address type if the address is of the address type
+        implemented by this class, otherewise returns None"""
+        if self._pattern.match(address):
+            return self.address_type
+        return None
+
+    def get_addresser(self, address):
+        """Returns the self if the address is of the address type
+        implemented by this class, otherewise returns None"""
+        if self._pattern.match(address):
+            return self
+        return None
+
+    def get_object_id(self, address):
+        """Returns the hash of an object_id (or the object_id itself if it
+        is a 12-byte unique identifier), as encoded in a given address"""
+        return address[14:38]
+
+    def get_target_id(self, address):
+        """Returns the hash of a target_id (or the target_id itself if it
+        is a 12-byte unique identifier), as encoded in a given address"""
```

Comment:
This comment doesn't line up with the behavior of the function.

Reply:
What specifically?

Reply:
"""Returns the hash of a target_id (or the target_id itself if it
        is a 12-byte unique identifier), as encoded in a given address"""

Yes, get_target_id does correctly return the hash of the target_id (or the target_id itself it is a 12-byte unique identifier), as encoded in the address.

    @pytest.mark.parse_role_member_address
    def test_parse_role_member_address_hash(self):
        """Test addresser.user.parse returns a parsed address if it is a user address"""
        user_id = "foo"
        role_id = "bar"
        address = addresser.role.member.address(role_id, user_id)
        parsed = addresser.role.member.parse(address)

        self.assertEqual(parsed.object_type, addresser.ObjectType.ROLE)
        self.assertEqual(parsed.related_type, addresser.ObjectType.USER)
        self.assertEqual(
            parsed.relationship_type, addresser.RelationshipType.MEMBER
        )
        self.assertEqual(parsed.address_type, addresser.AddressSpace.ROLES_MEMBERS)
        self.assertEqual(parsed.object_id, addresser.role.member.hash(role_id))
        self.assertEqual(parsed.target_id, addresser.role.member.hash(user_id))